lib/pkcs11h-serialization.c

```diff
@@ -334,12 +339,17 @@ pkcs11h_certificate_serializeCertificateId (
 		}
 
 		sz[n-1] = '/';
-		rv = _pkcs11h_util_binaryToHex (
-			sz+n,
-			saved_max-n,
-			certificate_id->attrCKA_ID,
-			certificate_id->attrCKA_ID_size
-		);
+		if (
+			(rv = _pkcs11h_util_binaryToHex (
+				sz+n,
+				saved_max-n,
+				certificate_id->attrCKA_ID,
+				certificate_id->attrCKA_ID_size,
+				__PKCS11H_SERIALIZE_AS_PKCS11_URI	
+			)) != CKR_OK
+		) {
+			goto cleanup;
+		}
```

Comment:
If we don't check the error code here, an unmodified "empty" target buffer will be returned in case `_pkcs11h_util_binaryToHex` fails.